src/engine/Settings.cs

```diff
@@ -379,7 +427,19 @@ private static Settings LoadSettings()
 
             file.Close();
 
-            return JsonConvert.DeserializeObject<Settings>(text);
+            try
+            {
+                return JsonConvert.DeserializeObject<Settings>(text);
+            }
+            catch
+            {
+                GD.Print("Settings configuration file is unreadable, using default settings.");
```

Comment:
And this error is actually the settings file has invalid content?

Reply:
It would trigger if the reading/deserializing failed for any reason, so I was treating it that way.

Reply:
But the error on reading is above, which I also commented on, right? So this is purely if the data is invalid that has been read.

Reply:
Technically the part above it actually reads the file.. I hadn't looked into whether var text = file.GetAsText(); can throw if the reading fails somehow.

Reply:
I think that can't throw as that's from Godot. The godot file open method doesn't fail with an exception.